### DISCUSSION TITLE
Auth: Hasura verify JWT And return response

### DISCUSSION THREAD

**Discussion by nizamani** [hasura/graphql-engine#8859] [General]
I want to ask can Hasura just validate a JWT without making an API call to database and return whether or not JWT is valid.
For example we make an API call to Hasura endpoint, can Hasura do this:
- tell us that JWT is valid for role user
- JWT hasn't expired
Or does Hasura only validates JWT when we make a graphql API call either to Hasura action(backend code) or fetching records from database?
This is because we want this to be done really fast so Hasura just tells us that JWT is valid or not valid without performing any API call to backend code or database.

If this feature is not already present in Hasura, for example Hasura can only make an API call and talk to either backend or database I would like to make this a feature request. This will be very useful feature on frontend where we just need to verify JWT from Hasura without having to hit database hence verification call will be much much faster. Right now if Hasura has to hit backend REST endpoint or hit our database this will make verification process too slow hence bad user experience for our customers

**Comment by ajohnson1200** [comment]
To the best of my knowledge this isn't possible OOTB, but on this:

> Right now if Hasura has to hit backend REST endpoint or hit our database this will make verification process too slow hence bad user experience for our customers

depending on what "too slow" is, it seems like you could create a computed field or some super simple query that would return in milliseconds.. what's slow for you? And what response time are you seeing from a simple Hasura query?
